### PR TITLE
PROD-2323 Add-tabs-to-display-monitored-and-ignored-schema-in-Data-Detection

### PR DESCRIPTION
**Pull request by lucanovera**
### Description Of Changes

In Data Detection, add tabs to display Monited, Ignored Schema and Action Required.

### Code Changes

* [ ] Added DetectionResultFilterTabs.tsx as a simple wrapper with styling changes for DataTabsHeader
* [ ] Added tabs to Data Detection & Discovery Page
* [ ] Added useDetectionResultsFilterTabs.tsx and useDiscoveryResultsFilterTabs.tsx, here I defined the tabs with their resulting diffStatus and childDiffStatus filtering. 
* [ ] Use changeTypeOverride prop to override the status icon and status column value to match the current tab (when viewing monitored tab everything should say monitored)
* [ ] Removed confusing eye icon for monitor/unmonitored

### Screenshots
Data Detection > Action Required Tab
Screen Shot 2024-08-28 at 10 49 14

Data Detection > Monitored Tab
Screen Shot 2024-08-28 at 10 49 20

Data Detection > Unmonitored Tab
Screen Shot 2024-08-28 at 10 49 23

Data Discovery > Action Required Tab
Screen Shot 2024-08-28 at 10 49 27

Data Discovery > Unmonitored Tab
Screen Shot 2024-08-28 at 10 49 40

### Steps to Confirm

* [ ] Go to the Data Detection page
* [ ] Check that using the Monitored tab it displays the Monitored status items only and that the icon for the changeType is monitored (green dot). Also the status column should read Monitored and the action column show only have the Ignore button.
* [ ] Check that using the Unmonitored tab it displays the Unmonitored status items only and that the icon for the changeType is unmonitored (red dot). Also the status column should read Unmonitored and the action column show only have the Monitor button.
* [ ] Check that you can navigate up and down the datasets while keeping the tab filter the same

* [ ] Go to the Data Disocvery page
* [ ] Check that using the Unmonitored tab it displays the Unmonitored status items only and that the icon for the changeType is unmonitored (red dot). Also the status column should read Unmonitored and the action column show no actions (decided best not to try to add this now, since there's a new UI version of DnD coming soon that solves these problems)
* [ ] Check that you can navigate up and down the datasets while keeping the tab filter the same

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by gilluminate** [comment]
At 1280 wide, these buttons get squished. Likely not caused by your work here, but something to address while we're here maybe?

CleanShot 2024-08-28 at 09 44 52@2x

**Comment by lucanovera** [comment]
> At 1280 wide, these buttons get squished. Likely not caused by your work here, but something to address while we're here maybe?



> At 1280 wide, these buttons get squished. Likely not caused by your work here, but something to address while we're here maybe?
> 
> CleanShot 2024-08-28 at 09 44 52@2x

Good catch. I just pushed a change adding a column width of 180 for the action column. It shouldn't shrink past that now.

File: clients/admin-ui/src/features/data-discovery-and-detection/tables/DetectionResultTable.tsx [hunk 1]
```diff
@@ -66,27 +68,18 @@ const DetectionResultTable = ({ resourceUrn }: MonitorResultTableProps) => {
...
+    setFilterTabIndex,
+    filterTabIndex,
+    activeDiffFilters,
+    activeChildDiffFilters,
+    activeChangeTypeOverride,
+  } = useDetectionResultFilterTabs({
+    initialFilterTabIndex: router.query?.filterTabIndex
+      ? Number(router.query?.filterTabIndex)
+      : undefined,
+  });
```
Comment (gilluminate) [inline clients/admin-ui/src/features/data-discovery-and-detection/tables/DetectionResultTable.tsx:82]: 👍 love the clean up

**Review by gilluminate — APPROVED** [review]
Looks great, overall. Just some minor clean up and possibly looking into that button issue.

**Pull request merged**